macros/models/_project/balancer/balancer_protocol_revenue_macro.sql

```diff
@@ -0,0 +1,148 @@
+            token AS token_address,
+            SUM(protocol_fees) AS protocol_fee_amount_raw
+        FROM {{ source('balancer_v2_' + blockchain, 'Vault_evt_PoolBalanceChanged') }} b
+        CROSS JOIN unnest("protocolFeeAmounts", "tokens") AS t(protocol_fees, token)
+        GROUP BY 1, 2, 3 
+
+        UNION ALL          
+
+        SELECT
+            date_trunc('day', t.evt_block_time) AS day,
+            poolId AS pool_id,
+            b.poolAddress AS token_address,
+            sum(value) AS protocol_fee_amount_raw
+        FROM {{ source('balancer_v2_' + blockchain, 'Vault_evt_PoolRegistered') }} b
+        JOIN {{ source('erc20_' + blockchain, 'evt_transfer') }} t
+            ON t.contract_address = b.poolAddress
+            AND t."from" = 0x0000000000000000000000000000000000000000
+            AND t.to = 0xce88686553686DA562CE7Cea497CE749DA109f9F --ProtocolFeesCollector address, which is the same across all chains
+        GROUP BY 1, 2, 3
+    ),
+
+    decorated_protocol_fee AS (
+        SELECT 
+            d.day, 
+            d.pool_id, 
+            d.token_address, 
+            SUM(d.protocol_fee_amount_raw) AS token_amount_raw, 
+            SUM(d.protocol_fee_amount_raw / power(10, COALESCE(t.decimals,p1.decimals))) AS token_amount,
+            CASE 
+                WHEN BYTEARRAY_SUBSTRING(d.pool_id, 1, 20) = d.token_address -- fees paid in BPTs
+                    THEN 0
+                ELSE 
+                    SUM(COALESCE(p1.price, p2.price, p3.price) * protocol_fee_amount_raw / POWER(10, COALESCE(t.decimals,p1.decimals))) 
+            END AS protocol_fee_collected_usd
+        FROM daily_protocol_fee_collected d
+        LEFT JOIN prices p1
+            ON p1.token = d.token_address
+            AND p1.day = d.day
+        LEFT JOIN dex_prices p2
+            ON p2.token = d.token_address
+            AND p2.day = d.day
+        LEFT JOIN bpt_prices p3
+            ON p3.token = CAST(d.token_address AS VARCHAR)
+            AND p3.day = d.day
+        LEFT JOIN {{ ref('tokens_erc20') }} t 
+            ON t.contract_address = d.token_address
+            AND t.blockchain = '{{blockchain}}'
+        GROUP BY 1, 2, 3
+    ),
+
+    revenue_share as(
+        SELECT
+        day,
+        CASE 
+            WHEN day < DATE '2022-07-03' THEN 1
+            WHEN day >= DATE '2022-07-03' AND day < DATE '2023-01-23' THEN 0.25
+            WHEN day >= DATE '2023-01-23' AND day < DATE '2023-07-24' THEN 0.35
+            WHEN day >= DATE '2023-07-24' THEN 0.175
+        END AS treasury_share
+    FROM UNNEST(SEQUENCE(DATE '2022-01-01', CURRENT_DATE, INTERVAL '1' DAY)) AS date(day)
+    )
+
+
+    SELECT
+        f.day,
+        f.pool_id,
+        BYTEARRAY_SUBSTRING(f.pool_id,1,20) as pool_address,
+        l.name AS pool_symbol,
+        '{{blockchain}}' as blockchain,
+        f.token_address,
+        SUM(f.token_amount_raw) as token_amount_raw,
+        SUM(f.token_amount) as token_amount,
```

Comment:
I'm re-thinking the need of having protocol fee by token since most pools pay the protocol fee in only one token. Also, if we have this breakdown here it means we'll always start queries for analyzing daily protocol fee collection by pool with a CTE:

`SELECT day, SUM(collected_usd), SUM(revenue_usd) FROM ..`

I'm curious to hear your guys thoughts

Reply:
removing token address as a field, do you think we should also remove token_amount_raw and token_amount or rename them as fee_amount_raw and fee_amount?

Reply:
Also, even considering that most pools pay the protocol fee in only one token, wouldn't it be a useful information to have (in which token protocol fees are paid in that pool, daily protocol fee by token etc)? We could maybe filter only for the cases where token_amount_raw > 0

Reply:
yea I'm fine about having the token to allow that kind of analysis - let's wrap this up and ask Dune team for review